dev/io.openliberty.microprofile.openapi.2.0.internal/src/io/openliberty/microprofile/openapi20/merge/MergeProcessor.java

```diff
@@ -0,0 +1,773 @@
+     * <p>
+     * Actually assumes that equals is implemented properly and just checks that the first element is equal to all others
+     * <p>
+     * If {@code collection} contains less than two elements, this method will always return {@code true}.
+     * 
+     * @param <T> the element type
+     * @param collection the collection of elements to test for equality
+     * @param comparator the function to use to test equality
+     * @return {@code true} if all elements of {@code collection} are equal, {@code false} otherwise
+     */
+    private static <T> boolean allEqual(Collection<? extends T> collection, BiPredicate<? super T, ? super T> comparator) {
+        Iterator<? extends T> i = collection.iterator();
+        if (!i.hasNext()) {
+            return true;
+        }
+
+        T first = i.next();
+        while (i.hasNext()) {
+            if (!equals(first, i.next(), comparator)) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     * Tests if two objects are equal, using {@code comparator} to test their equality if both {@code a} and {@code b} are not {@code null}.
+     * 
+     * @param <T> the type of {@code a} and {@code b}
+     * @param a the first object
+     * @param b the second object
+     * @param comparator the comparison function
+     * @return {@code true} if {@code a} and {@code b} are equal, {@code false} otherwise
+     */
+    private static <T> boolean equals(T a, T b, BiPredicate<? super T, ? super T> comparator) {
+        if (a == null) {
+            return b == null ? true : false;
+        } else {
+            return b == null ? false : comparator.test(a, b);
+        }
+    }
+
+    /**
+     * Converts {@code null} to an empty map
+     * 
+     * @param in a map, or {@code null}
+     * @return an empty map if {@code in} is {@code null}, otherwise {@code in}
+     */
+    private static <K, V> Map<K, V> notNull(Map<K, V> in) {
+        if (in == null) {
+            return Collections.emptyMap();
+        } else {
+            return in;
+        }
+    }
+
+    /**
+     * Converts {@code null} to an empty list
+     * 
+     * @param in a list, or {@code null}
+     * @return an empty list if {@code in} is {@code null}, otherwise {@code in}
+     */
+    private static <V> List<V> notNull(List<V> in) {
+        if (in == null) {
+            return Collections.emptyList();
+        } else {
+            return in;
+        }
+    }
+
+    /**
+     * Creates and stores new unique names for model elements.
+     * <p>
+     * Certain names must be unique within an OpenAPI document. When merging documents, these names may have to be changed so that they don't clash.
+     * <p>
+     * This class handles detecting clashes, creating new names as required and looking up the new names of renamed elements.
+     */
+    public class RenameHolder {
```

Comment:
In general the MergeProcessor class seems a little long and there are things that could be moved ... RenameHolder could be moved to its own standalone class, although maybe with a better name?

Reply:
If I wasn't suggesting moving this out, I would be wondering if RenameHolder should be a private static class?

Reply:
I don't disagree with a better name :stuck_out_tongue_closed_eyes: 

It currently has to be a non-static inner class because it references `MergeProcessor.namesInUse`.

Possibly we could move it out into a separate file, but it would still have to be an outer and an inner class, with the outer class owning `namesInUse`.

Reply:
Moved `RenameHolder` into a different file.

Now have `NameProcessor` which holds `namesInUse` with an inner class `DocumentNameProcessor`.